src/shared/components/post/PostFooter.tsx

```diff
@@ -3,19 +3,40 @@ import { Box, IconButton, Typography } from "@mui/material";
 import { FavoriteBorder, MoreHoriz, Favorite } from "@mui/icons-material/";
 import useTheme from "features/theme/useTheme";
 import { BasePostFooterType } from "shared/types";
+import { useSelector } from "react-redux";
+import { getCurrentUser } from "features/auth/authSlice";
+import axios from "axios";
 
-function PostFooter({ id, userId, likes, comments }: BasePostFooterType) {
+axios.defaults.withCredentials = true;
+
+function PostFooter({ id, likes, comments }: BasePostFooterType) {
   const {
     themeColor: { color, navButtons },
   } = useTheme();
+  const user = useSelector(getCurrentUser);
   const [showComment, setShowComment] = useState(false);
   const [displayLikeCount, setDisplayLikeCount] = useState(likes.length);
-  const [isLiked, setIsLiked] = useState(false);
+  const [isLiked, setIsLiked] = useState(
```

Comment:
unneccessary state here. liked is a prop so this comment will rerender with the new likes array.
you can just do
const isLiked = likes.find(like => like.id === user.id)

Reply:
Ah, I see. I currently only have the PostFooter handling all the likes/dislikes, but I do see how the props can be used to rerender the like button.